src/content/en/showcase/2017/santa.md

```diff
@@ -0,0 +1,217 @@
+
+## Offline Design
+
+Santa Tracker is already neatly segmented into scenes thanks to Polymer and `lazy-pages`; plus each scene has its own directory.
+We designed Santa Tracker's Service Worker, the core piece that enables offline which runs on a user's browser, to be aware of the shared code vs 'scene' distinction.
+
+What's the theory behind Service Worker? When a user on a supported browser loads your site, the frontend HTML can request that the Service Worker (SW) is installed.
+For Santa Tracker, the SW lives at [`/sw.js`](https://santatracker.google.com/sw.js).
+This then runs an 'install' event that will precache all of Santa's shared code, so nothing needs to be fetched at runtime.
+
+<figure>
+  <img src="/web/showcase/2017/images/santa/sw-flow.png"  />
+</figure>
+
+The Service Worker, once installed, is able to intercept all HTTP requests.
+For Santa Tracker, the simplified decision flow looks like-
+
+1. Is the request already cached?
+  * Great! Return the cached response.
+2. Does the request being made match a scene directory, like "scenes/boatload/boatload-scene_en.html"?
+  * Perform a network request, and store the result in the cache before returning it to the user.
+3. Otherwise, perform a regular network request.
+
+Our flow and 'install' event allows Santa Tracker to load, even while a user is offline.
+However, only the scenes a user has previously loaded will be available—perfect for replaying that one game and beating your high score.
+
+Keen observers may note that our caching strategy doesn't allow for _changes_ in content.
+Once a file is cached in a user's browser, it will never be changed.
+More on that later!
+
+### We'll Do It Live
+
+As we mentioned, Santa's elves also hard at work throughout December, and they often have to release new updates throughout the month.
+When a release of Santa Tracker is built, it is given a unique label—e.g., `v20161204112055`, the timestamp of the release (11:20:55 on the 4th of December, 2016).
+
+For this labelled release, we generate a MD5 hash of every file and store this in our 'cache manifest'.
+On a modern SSD, this only adds a few seconds.
+
+Each release is deployed to a unique path on Google's static caching server—that is, older releases are never removed.
+This means that after a new release, all assets will have a different URL—even if they didn't change—and anything cached by the browser or the service worker will be useless, unless we do some extra work.
+
+We also deploy a new version of what we call the "prod" resources—Santa's index HTML and Service Worker, which live on [https://santatracker.google.com/](https://santatracker.google.com/)—this does overwrite the old version.
+
+<figure>
+  <img src="/web/showcase/2017/images/santa/static-prod.png"  />
+</figure>
+
+Whenever Santa Tracker is loaded, clients will always refetch the Service Worker, to see if there's an upgrade available.
```

Comment:
Try:

Whenever Santa Tracker loads, the browser will check for an updated service worker and fetch it, if available.